src/python/tritonfrontend/_c/tritonfrontend.h

```diff
@@ -0,0 +1,143 @@
+};
+struct InvalidArgumentError : public TritonError {
+  explicit InvalidArgumentError(const std::string& what) : TritonError(what) {}
+};
+struct UnavailableError : public TritonError {
+  explicit UnavailableError(const std::string& what) : TritonError(what) {}
+};
+struct UnsupportedError : public TritonError {
+  explicit UnsupportedError(const std::string& what) : TritonError(what) {}
+};
+struct AlreadyExistsError : public TritonError {
+  explicit AlreadyExistsError(const std::string& what) : TritonError(what) {}
+};
+
+void
+ThrowIfError(TRITONSERVER_Error* err)
+{
+  if (err == nullptr) {
+    return;
+  }
+  std::shared_ptr<TRITONSERVER_Error> managed_err(
+      err, TRITONSERVER_ErrorDelete);
+  std::string msg = TRITONSERVER_ErrorMessage(err);
+  switch (TRITONSERVER_ErrorCode(err)) {
+    case TRITONSERVER_ERROR_INTERNAL:
+      throw InternalError(std::move(msg));
+    case TRITONSERVER_ERROR_NOT_FOUND:
+      throw NotFoundError(std::move(msg));
+    case TRITONSERVER_ERROR_INVALID_ARG:
+      throw InvalidArgumentError(std::move(msg));
+    case TRITONSERVER_ERROR_UNAVAILABLE:
+      throw UnavailableError(std::move(msg));
+    case TRITONSERVER_ERROR_UNSUPPORTED:
+      throw UnsupportedError(std::move(msg));
+    case TRITONSERVER_ERROR_ALREADY_EXISTS:
+      throw AlreadyExistsError(std::move(msg));
+    default:
+      throw UnknownError(std::move(msg));
+  }
+}
+
+
+template <typename Base, typename FrontendServer>
+class TritonFrontend {
+ private:
+  std::shared_ptr<TRITONSERVER_Server> server_;
+  std::unique_ptr<Base> service;
+  triton::server::RestrictedFeatures restricted_features;
+  // TODO: [DLIS-7194] Add support for TraceManager & SharedMemoryManager
+  // triton::server::TraceManager trace_manager_;
+  // triton::server::SharedMemoryManager shm_manager_;
+
+ public:
+  TritonFrontend(uintptr_t server_mem_addr, UnorderedMapType data)
+  {
+    TRITONSERVER_Server* server_ptr =
+        reinterpret_cast<TRITONSERVER_Server*>(server_mem_addr);
+
+    server_.reset(server_ptr, EmptyDeleter);
+
+    ThrowIfError(FrontendServer::Create(
+        server_, data, nullptr /* TraceManager */,
+        nullptr /* SharedMemoryManager */, restricted_features, &service));
+  };
+
+  // TODO: [DLIS-7194] Add support for TraceManager & SharedMemoryManager
+  // TritonFrontend(
+  //     uintptr_t server_mem_addr, UnorderedMapType data,
+  //     TraceManager trace_manager, SharedMemoryManager shm_manager)
+
+  void StartService() { ThrowIfError(service->Start()); };
+  void StopService() { ThrowIfError(service->Stop()); };
+
+
+  static TRITONSERVER_Error* EmptyDeleter(TRITONSERVER_Server* obj)
+  {
+    // The frontend does not own the TRITONSERVER_Server* object.
+    // Hence, deleting the underlying server instance,
+    // will cause a double-free when the core bindings attempt to
+    // delete the TRITONSERVER_Server instance.
+    return nullptr;
+  };
```

Comment:
I don't think the destructor should be returning anything - it would likely get lost or unused if this were later changed to return an actual error object.

Can probably be something like this:
```cpp
    // The frontend does not own the TRITONSERVER_Server* object.
    // Hence, deleting the underlying server instance,
    // will cause a double-free when the core bindings attempt to
    // delete the TRITONSERVER_Server instance.
    static void EmptyDeleter(TRITONSERVER_Server* obj) {};
```

Reply:
Changes Implemented